### DISCUSSION TITLE
Batch Extracting Dominant Color from an Image

### DISCUSSION THREAD

**Discussion by sheedyaja** [ImageMagick/ImageMagick#2776] [Help] [answered]
Hi

I'm trying to extract color code from my images (18k-ish images in total)
So far I use this command
`magick convert *.jpg -resize 1x1 @out.txt`
Which yield this output

```
# ImageMagick pixel enumeration: 1,1,65535,srgb
0,0: (232.704,232.791,232.128)  #E9E9E8  srgb(91.2563%,91.2906%,91.0306%)
# ImageMagick pixel enumeration: 1,1,65535,srgb
0,0: (234.715,233.527,229.405)  #EBEAE5  srgb(92.045%,91.579%,89.9626%)
# ImageMagick pixel enumeration: 1,1,65535,srgb
0,0: (225.225,216.245,197.354)  #E1D8C5  srgb(88.3235%,84.802%,77.3938%)
# ImageMagick pixel enumeration: 1,1,65535,srgb
0,0: (194.39,183.679,167.428)  #C2B8A7  srgb(76.2312%,72.031%,65.6582%)
```

I need the output to also state which file name and which color hex code, and only hex code in text file
It will be better if the color code is in HSL format

Any help? 
Thanks

**Comment by sheedyaja** [comment, marked as answer]
Alright, I found a solution
magick convert *.jpg -resize 1x1 -format "%f,%[fx:int(mean.r*255)],%[fx:int(mean.g*255)],%[fx:int(mean.b*255)]\n" info:

Convert the result to HSL, and set the color name from the color range

**Reply by fmw42** [reply]
For IM 7, use magick, not magick convert. 

For Hex, try

`magick *.jpg -resize 1x1! -format "%f,%[hex]\n" info: > textfile.txt`

For HSL

`magick *.jpg -resize 1x1! -colorspace HSL -format "%f,%[pixel:u.p{0,0}]\n" info: > textfile.txt`

**Reply by sheedyaja** [reply]
Thanks, it's faster and better 

Now I need a way to convert the HSL to color name

**Reply by fmw42** [reply]
Not all possible colors have a color name. You can make a table between rgb and names for those named colors. See https://imagemagick.org/script/color.php#color_names

**Comment by snibgo** [comment]
I suggest you use ! in resize, eg "-resize 1x1!". Otherwise, input images that have large aspect ratios will result in more than one pixel.

After resizing, you can "-colorspace HSL" if you want.

**Reply by sheedyaja** [reply]
okay, tried it and it's better in result
Thanks
